packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -116,39 +127,85 @@ export const searchItems = ( items = [], searchTerm = '', config = {} ) => {
 		return items;
 	}
 
-	const defaultGetTitle = ( item ) => item.title;
-	const defaultGetKeywords = ( item ) => item.keywords || [];
-	const defaultGetCategory = ( item ) => item.category;
+	const rankedItems = items
+		.map( ( item ) => {
+			return [ item, getItemSearchRank( item, searchTerm, config ) ];
+		} )
+		.filter( ( [ , rank ] ) => rank > 0 );
+
+	rankedItems.sort( ( [ , rank1 ], [ , rank2 ] ) => rank2 - rank1 );
+	return rankedItems.map( ( [ item ] ) => item );
+};
+
+/**
+ * Get the search rank for a given iotem and a specific search term.
+ * The higher is higher for items with the best match.
+ * If the rank equals 0, it should be excluded from the results.
+ *
+ * @param {Object} item       Item to filter.
+ * @param {string} searchTerm Search term.
+ * @param {Object} config     Search Config.
+ * @return {number}           Search Rank.
+ */
+export function getItemSearchRank( item, searchTerm, config = {} ) {
+	const defaultGetName = ( it ) => it.name || '';
+	const defaultGetTitle = ( it ) => it.title;
+	const defaultGetKeywords = ( it ) => it.keywords || [];
+	const defaultGetCategory = ( it ) => it.category;
 	const defaultGetCollection = () => null;
 	const defaultGetVariations = () => [];
+
 	const {
+		getName = defaultGetName,
 		getTitle = defaultGetTitle,
 		getKeywords = defaultGetKeywords,
 		getCategory = defaultGetCategory,
 		getCollection = defaultGetCollection,
 		getVariations = defaultGetVariations,
 	} = config;
 
-	return items.filter( ( item ) => {
-		const title = getTitle( item );
-		const keywords = getKeywords( item );
-		const category = getCategory( item );
-		const collection = getCollection( item );
-		const variations = getVariations( item );
+	const name = getName( item );
+	const title = getTitle( item );
+	const keywords = getKeywords( item );
+	const category = getCategory( item );
+	const collection = getCollection( item );
+	const variations = getVariations( item );
+
+	const sanitizedSearchTerm = sanitizeTerm( searchTerm );
+	const sanitizedTitle = sanitizeTerm( title );
+
+	let rank = 0;
 
+	// Prefers exact matchs
+	// Then prefers if the beginning of the title matches the search term
+	// Keywords, categories, collection, variations match come later.
+	if ( sanitizedSearchTerm === sanitizedTitle ) {
+		rank += 30;
+	} else if ( sanitizedTitle.indexOf( sanitizedSearchTerm ) === 0 ) {
+		rank += 20;
+	} else {
 		const terms = [
 			title,
 			...keywords,
 			category,
 			collection,
 			...variations,
 		].join( ' ' );
-
+		const normalizedSearchTerms = words( sanitizedSearchTerm );
```

Comment:
(minor, subjective) Maybe this becomes cleaner if the input before piping through `words` is called _input_ rather than _term_:

```suggestion
		const normalizedSearchTerms = words( normalizedSearchInput );
```

Reply:
I feel if I do this change, i need to rename a bunch of things (even untouched) on this file like the normalizeSearchTerm functions...

Reply:
I went ahead and renamed a bunch of things, let me know if it's better.

Reply:
Thanks, looks great.